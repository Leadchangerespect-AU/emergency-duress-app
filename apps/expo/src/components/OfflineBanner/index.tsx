import React from "react";
import { Text, View } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import colors from "tailwindcss/colors";

type OfflineBannerProps = {
  isVisible: boolean;
};

const OfflineBanner = ({ isVisible }: OfflineBannerProps) => {
  if (!isVisible) return null;

  return (
    <View className="flex-row items-center justify-center bg-yellow-500 p-2">
      <Ionicons name="cloud-offline" size={18} color={colors.black} />
      <Text className="text-back ml-2 text-center">
        You are offline. Some features may be limited.
      </Text>
    </View>
  );
};

export default OfflineBanner;
