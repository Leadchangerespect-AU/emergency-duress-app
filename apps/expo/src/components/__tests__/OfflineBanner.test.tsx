import React from "react";
import { render, screen } from "@testing-library/react-native";

import OfflineBanner from "../OfflineBanner";

jest.mock("@expo/vector-icons", () => ({
  Ionicons: "",
}));

describe("OfflineBanner", () => {
  it("should not render when isVisible is false", () => {
    render(<OfflineBanner isVisible={false} />);

    // The component should return null when not visible
    expect(
      screen.queryByText("You are offline. Some features may be limited."),
    ).toBeNull();
  });

  it("should render when isVisible is true", () => {
    render(<OfflineBanner isVisible={true} />);

    // The component should display the offline message
    expect(
      screen.getByText("You are offline. Some features may be limited."),
    ).toBeTruthy();
  });

  it("should have the correct styling", () => {
    render(<OfflineBanner isVisible={true} />);

    // Check that the text has the correct styling
    const textElement = screen.getByText(
      "You are offline. Some features may be limited.",
    );
    expect(textElement.props.className).toContain("text-back ml-2 text-center");

    // The parent View should have the yellow background
    const parentView = textElement.parent.parent;
    expect(parentView.props.className).toContain("bg-yellow-500");
  });

  it("matches snapshot", () => {
    render(<OfflineBanner isVisible />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
