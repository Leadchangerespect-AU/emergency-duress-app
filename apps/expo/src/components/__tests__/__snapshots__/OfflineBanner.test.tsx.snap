// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineBanner matches snapshot 1`] = `
<View
  className="flex-row items-center justify-center bg-yellow-500 p-2"
>
  <
    color="#000"
    name="cloud-offline"
    size={18}
  />
  <Text
    className="text-back ml-2 text-center"
  >
    You are offline. Some features may be limited.
  </Text>
</View>
`;
